cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1514,13 +1520,19 @@ void reader::impl::allocate_columns(read_mode mode, size_t skip_rows, size_t num
         // we're going to start null mask as all valid and then turn bits off if necessary
         out_buf.create_with_mask(
           out_buf.type.id() == type_id::LIST && l_idx < max_depth ? num_rows + 1 : num_rows,
-          cudf::mask_state::ALL_VALID,
+          cudf::mask_state::UNINITIALIZED,
+          false,
           _stream,
           _mr);
+        memset_bufs.push_back(cudf::device_span<std::byte>(static_cast<std::byte*>(out_buf.data()),
+                                                           out_buf.data_size()));
+        nullmask_bufs.push_back(cudf::device_span<uint32_t>(
```

Comment:
We should write this as `cudf::bitmask_type`. https://github.com/rapidsai/cudf/blob/cc19d8a7b424abbc87f7767e3bc60c54390dc9e3/cpp/include/cudf/types.hpp#L96C7-L96C25
```suggestion
        nullmask_bufs.push_back(cudf::device_span<cudf::bitmask_type>(
```